### PR TITLE
Update add_job and alter_job documentation for fixed schedules

### LINKED ISSUE #1791: [Docs RFC] Fixed schedules for background jobs
# Describe change in content, appearance, or functionality
 
Currently, the next start of a scheduled background job is computed by adding its `schedule_interval` to its finish time. This leads to the schedule "drifting" after each execution by the duration of the job, and users have been requesting the ability to set fixed schedules for their jobs. For example, having a job execute every Monday at midnight, instead of midnight, 2 am, 4 am, 6 am if the job takes 2 hours to run. 

We have now introduced the ability to schedule background jobs that will run on a fixed schedule. (Commit https://github.com/timescale/timescaledb/commit/54ed0d5c05df990674a6fc9857e1120cdcc9b338)
The parameter `initial_start` is used as the reference point and all subsequent executions of a job will be scheduled to run at multiples of `schedule_interval` added to `initial_start`. If initial_start is not provided, then the current timestamp is used.

Both behaviors are now supported, drifting and fixed. 
This "fixed schedule" behavior will be the default from now on for User Defined Actions. 
For policies, it must be specified explicitly that a fixed schedule is desired, by setting the newly-added parameter `initial_start`, otherwise the behavior defaults to the current drifting one. 

I hope that's a good overview of the changes, the linked design document contains more information and API reference / examples. 
# Subject matter expert (SME)

konskov

# Deadline

2.9.0

# Any further info

This [design doc here](https://docs.google.com/document/d/1KpCHFBMK5MZFXQTYqRu3cVMTORqgzPl9iMNQ8xwKgTU/edit#) has an API reference for fixed schedule jobs. 

## Contributing to documentation

We welcome documentation contributions!

* For information about how to suggest a change, see the [contributing guide](https://github.com/timescale/docs/blob/latest/CONTRIBUTING.md) in our GitHub repository.
* For information on style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs)

### PR DESCRIPTION
**Pull request by konskov**
# Description

Add documentation for fixed schedules for background jobs.

# Links

Fixes #1791 

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: api/alter_job.md [hunk 1]
```diff
@@ -45,6 +45,9 @@ prevents the job from attempting to be started again while it is running. When
 the job completes, whether or not the job is successful, the parameter is
 automatically updated to the next computed start time.
 
+Note that, the optional parameters `initial_start` and `timezone` are not modifiable with `alter_job`. It is not currently possible to alter these parameters for an existing job. If you wish to do so, you must drop the existing job and register it again with the new values. It is also not possible to change the value of the `fixed_schedule` parameter. You must drop and recreate the job if you wish to alter this behavior. 
+Also note that altering the `next_start` value is only effective for the next execution of the job in case of fixed schedules. On the next execution, it will automatically return to the schedule.
```
Comment (Loquacity) [inline api/alter_job.md:49]: Reflow

File: api/add_compression_policy.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ the policies that already exist, see
 |---|---|---|
 |`hypertable`|REGCLASS|Name of the hypertable or continuous aggregate|
 |`compress_after`|INTERVAL or INTEGER|The age after which the policy job compresses chunks. `compress_after` is calculated relative to the current time, so chunks containing data older than `now - {compress_after}::interval` are compressed.|
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval will be the interval between the finish time of the last execution and the next start. If provided, it will serve as the origin with respect to which the next_start is calculated |
+|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the compression policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing will be performed. Defaults to `NULL`.|
```
Comment (Loquacity) [inline api/add_compression_policy.md:33]:
```suggestion
|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the compression policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```

File: api/add_continuous_aggregate_policy.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ policies that you set or the policies that already exist, see
 |`start_offset`|INTERVAL or integer|Start of the refresh window as an interval relative to the time when the policy is executed. `NULL` is equivalent to `MIN(timestamp)` of the hypertable.|
 |`end_offset`|INTERVAL or integer|End of the refresh window as an interval relative to the time when the policy is executed. `NULL` is equivalent to `MAX(timestamp)` of the hypertable.|
 |`schedule_interval`|INTERVAL|Interval between refresh executions in wall-clock time. Defaults to 24 hours|
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval will be the interval between the finish time of the last execution and the next start. If provided, it will serve as the origin with respect to which the next_start is calculated |
+|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the refresh policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing will be performed. Defaults to `NULL`.|
```
Comment (Loquacity) [inline api/add_continuous_aggregate_policy.md:27]:
```suggestion
|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the refresh policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```

File: api/add_reorder_policy.md [hunk 4]
```diff
@@ -30,6 +30,8 @@ and re-create the policy if many older chunks have been affected.
 |---|---|---|
 | `hypertable` | REGCLASS | Hypertable to create the policy for. |
 | `index_name` | TEXT | Existing index by which to order rows on disk. |
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval will be the interval between the finish time of the last execution and the next start. If provided, it will serve as the origin with respect to which the next_start is calculated |
+|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the reorder policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing will be performed. Defaults to `NULL`.|
```
Comment (Loquacity) [inline api/add_reorder_policy.md:34]:
```suggestion
|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the reorder policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```

File: api/add_retention_policy.md [hunk 5]
```diff
@@ -22,6 +22,8 @@ one retention policy may exist per hypertable.
 |---|---|---|
 | `relation` | REGCLASS | Name of the hypertable or continuous aggregate to create the policy for. |
 | `drop_after` | INTERVAL or INTEGER | Chunks fully older than this interval when the policy is run are dropped|
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval will be the interval between the finish time of the last execution and the next start. If provided, it will serve as the origin with respect to which the next_start is calculated |
+|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the retention policy are aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing will be performed. Defaults to `NULL`.|
```
Comment (Loquacity) [inline api/add_retention_policy.md:26]:
```suggestion
|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the retention policy are aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```

File: api/alter_job.md (hunk 1 above)
Comment (konskov) [reply api/alter_job.md:49]: thank you @Loquacity for the quick review! Sorry, I did not understand what should change here wrt reflow? 😅

File: api/add_retention_policy.md (hunk 5 above)
Comment (konskov) [reply api/add_retention_policy.md]: Fixed

File: api/add_compression_policy.md (hunk 2 above)
Comment (konskov) [reply api/add_compression_policy.md]: Fixed

File: api/add_continuous_aggregate_policy.md (hunk 3 above)
Comment (konskov) [reply api/add_continuous_aggregate_policy.md]: fixed

File: api/add_reorder_policy.md (hunk 4 above)
Comment (konskov) [reply api/add_reorder_policy.md]: fixed

File: api/add_job.md [hunk 6]
```diff
@@ -48,6 +50,16 @@ END
...
 ```
 
+Register the `user_defined_action` procedure to run at midnight every Sunday:
+The `initial_start` provided must satisfy these requirements, so it must be a Sunday midnight.
+
+```sql
+-- December 4, 2022 is a Sunday
+SELECT add_job('user_defined_action','1 week', initial_start => '2022-12-04 00:00:00+00'::timestamptz);
+-- if subject to DST
+SELECT add_job('user_defined_action','1 week', initial_start => '2022-12-04 00:00:00+00'::timestamptz, timezone => 'Europe/Berlin');
```
Comment (konskov) [inline api/add_job.md:63]: not sure why these examples are not displayed, so I need some pointers here on what else I should change

File: api/alter_job.md (hunk 1 above)
Comment (Loquacity) [reply api/alter_job.md:49]: We hard wrap lines at 80 chars, so it just needs to be wrapped, that's all 👍

**Review by Loquacity — APPROVED** [review]

File: api/add_reorder_policy.md [hunk 7]
```diff
@@ -30,6 +30,8 @@ and re-create the policy if many older chunks have been affected.
 |---|---|---|
 | `hypertable` | REGCLASS | Hypertable to create the policy for. |
 | `index_name` | TEXT | Existing index by which to order rows on disk. |
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval is the intervalbetween the finish time of the last execution and the next start. If provided, it serves as the origin with respect to which the next_start is calculated |
```
Comment (Loquacity) [inline api/add_reorder_policy.md:33]:
```suggestion
|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval is the interval between the finish time of the last execution and the next start. If provided, it serves as the origin with respect to which the next_start is calculated |
```

File: api/add_reorder_policy.md [hunk 8]
```diff
@@ -30,6 +30,8 @@ and re-create the policy if many older chunks have been affected.
...
+|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the reorder policy will be aligned on its initial start. However, daylight savings time (DST) changes may shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```
Comment (Loquacity) [inline api/add_reorder_policy.md:34]:
```suggestion
|`timezone`|TEXT|A valid time zone. If `initial_start` is also specified, subsequent executions of the reorder policy are aligned on its initial start. However, daylight savings time (DST) changes might shift this alignment. Set to a valid time zone if this is an issue you want to mitigate. If omitted, UTC bucketing is performed. Defaults to `NULL`.|
```

File: api/add_retention_policy.md [hunk 9]
```diff
@@ -22,6 +22,8 @@ one retention policy may exist per hypertable.
 |---|---|---|
 | `relation` | REGCLASS | Name of the hypertable or continuous aggregate to create the policy for. |
 | `drop_after` | INTERVAL or INTEGER | Chunks fully older than this interval when the policy is run are dropped|
+|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval is the intervalbetween the finish time of the last execution and the next start. If provided, it serves as the origin with respect to which the next_start is calculated |
```
Comment (Loquacity) [inline api/add_retention_policy.md:25]:
```suggestion
|`initial_start`|TIMESTAMPTZ|Time the policy is first run. Defaults to NULL. If omitted, then the schedule interval is the interval between the finish time of the last execution and the next start. If provided, it serves as the origin with respect to which the next_start is calculated |
```

**Review by Loquacity — COMMENTED** [review]
A few things to make the linter happier

**Review by iroussos — APPROVED** [review]

**Review by Loquacity — APPROVED** [review]
I think we've got everything now!

**Comment by iroussos** [comment]
Thank you @Loquacity!

**Pull request merged**